### PR TITLE
XSUP-44190/SplunkPy

### PR DESCRIPTION
**Pull request by YaelShamai**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-44190

## Description
- Fixed an issue where **splunk-submit-event-hec** command failed when event included a single quotation mark.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/SplunkPy/ReleaseNotes/3_1_47.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#### Integrations
+
+##### SplunkPy
+
+- Fixed an issue where **splunk-submit-event-hec** command failed when event included a single quotation mark.
```
Comment (ShirleyDenkberg) [inline Packs/SplunkPy/ReleaseNotes/3_1_47.md:5]:
```suggestion
Fixed an issue where the ***splunk-submit-event-hec*** command failed when an event included a single quotation mark.
```

**Comment by ShirleyDenkberg** [comment]
@ilappe @JasBeilin Doc review completed.

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 2]
```diff
@@ -2667,27 +2667,11 @@ def parse_fields(fields):
...
-        events_str = str(events)
-
-        events_str = events_str.replace("'", '"')
-        rgx = re.compile(r"}[\s]*{")
-        valid_json_events = rgx.sub("},{", events_str)
-        valid_json_events = json.loads(f"[{valid_json_events}]")
-        return valid_json_events
-    except Exception as e:
-        raise DemistoException(f'{str(e)}\nMake sure that the events are in the correct format.')
+def extract_indexes(events: str | dict):
```
Comment (JasBeilin) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2670]: Please add a dostring

File: Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py [hunk 3]
```diff
@@ -2667,27 +2667,11 @@ def parse_fields(fields):
...
+    events_str = str(events)
+    rgx = r"""["'][\s]*index[\s]*["'][\s]*:[\s]*["']([^"']+)["']"""
```
Comment (JasBeilin) [inline Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py:2672]: Lets make that a Constant

**Review by JasBeilin — APPROVED** [review]
Nice!

**Pull request merged**